pkgs/development/tools/misc/ccache/default.nix

```diff
@@ -2,11 +2,11 @@
 
 let ccache = stdenv.mkDerivation rec {
   pname = "ccache";
-  version = "3.4.1";
+  version = "3.7.10";
 
   src = fetchurl {
-    sha256 = "1pppi4jbkkj641cdynmc35jaj40jjicw7gj75ran5qs5886jcblc";
-    url = "mirror://samba/ccache/${pname}-${version}.tar.xz";
+    sha256 = "0a7dajdpnh14nmd10ji897rzqn6847lr36hdbig2rv7sxbc6d58c";
+    url = "https://github.com/ccache/${pname}/releases/download/v${version}/${pname}-${version}.tar.xz";
   };
```

Comment:
Since the source is on GitHub now, we can use `fetchFromGitHub`:

```nix
  src = fetchFromGitHub {
    owner = pname;
    repo = pname;
    rev = "v${version}";
    sha256 = "0v6pvj50y18fxh2f6cx6xyg545xcpxfmrcg9bzbc6lc5acrvvjm7";
  };
```

`fetchFromGitHub` should also be added as an argument to the derivation at the top. `fetchurl` can be removed, since it is then not used anymore.

Reply:
I'm using `fetchFromGitHub` now, which required adding a `asciidoc` and `gperf` build-time dependencies. I assume those come pre-processed when using a source distribution.